backend/dataall/api/Objects/DatasetProfiling/resolvers.py

```diff
@@ -141,7 +146,27 @@ def get_profiling_results_from_s3(environment, dataset, table, run):
 
 
 def list_table_profiling_runs(context: Context, source, tableUri=None):
+    """
+    Lists the runs of a profiling job on a Table.
+    For datasets "Unclassified" all users can perform this action.
+    For datasets "Secret" or "Official", only users with PREVIEW_DATASET_TABLE permissions can perform this action.
+    """
     with context.engine.scoped_session() as session:
+        table: models.DatasetTable = db.api.DatasetTable.get_dataset_table_by_uri(
+            session, tableUri
+        )
+        dataset = db.api.Dataset.get_dataset_by_uri(session, table.datasetUri)
+        if (
+                dataset.confidentiality
+                != models.ConfidentialityClassification.Unclassified.value
+        ):
+            ResourcePolicy.check_user_resource_permission(
+                session=session,
+                username=context.username,
+                groups=context.groups,
+                resource_uri=table.tableUri,
```

Comment:
Hm, I thought that we set permission on dataset level. For example, if we want to update table in `DatasetTable#update_dataset_table` we pass dataset uri

Reply:
I am also double checking this as I did not implement the table preview checks. I also think that we need to add some tests to test the "secret" unauthorized datasets.

Reply:
I can confirm that the permission `PREVIEW_DATASET_TABLE` is applied on tables and is added at the creation of the table as `DATASET_TABLE_READ` 
```
            ResourcePolicy.attach_resource_policy(
                session=session,
                group=group,
                permissions=permissions.DATASET_TABLE_READ,
                resource_uri=table.tableUri,
                resource_type=models.DatasetTable.__name__,
            )
```

I have added several tests in my last commit. I think the PR is ready for a re-review :)

Reply:
Thanks for the update and checking it!